src/zenml/materializers/pandas_materializer.py

```diff
@@ -77,7 +82,36 @@ def load(self, data_type: Type[Any]) -> Union[pd.DataFrame, pd.Series]:
         Returns:
             The pandas dataframe or series.
         """
-        if fileio.exists(self.parquet_path):
+        # If the metadata file exists, then the data is stored as a chunked
+        #  parquet file in the latest version of the materializer
+        if fileio.exists(self.metadata_path):
+            if self.pyarrow_exists:
+                dfs = []
+
+                # Read the length of the dataframe from the metadata file
+                len_df = int(
+                    io_utils.read_file_contents_as_string(self.metadata_path)
+                )
+
+                for i in range(0, len_df, CHUNK_SIZE):
+                    with fileio.open(
+                        f"{self.parquet_path}_{i//CHUNK_SIZE}", mode="rb"
+                    ) as f:
+                        dfs.append(pd.read_parquet(f))
+
+                # concatenate all the dataframes to one
+                df = pd.concat(dfs)
```

Comment:
does `df.concat(dfs)` uses memory? will it be a problem in the case of a very large dataframe?

Reply:
Hmm wouldn’t be more of a problem than what was currently status quo :)

Reply:
@coderabbitai what do you think? is there a better way of doing this?

Reply:
No guys we dont need to touch this, of course we have to hold the dataframe in memory... thats what materialization means right?